src/pe/utils.rs

```diff
@@ -97,6 +97,16 @@ pub fn find_offset(
                 section.virtual_address,
                 section.virtual_address + section.virtual_size
             );
+            if file_alignment < 512
```

Comment:
1. The usage of `file_alignment` in the implementation of `section_read_size()` only requires it to be a strictly positive power of two. The additional restrictions *promised* by the Windows documentation do not need to be fulfilled in order for `section_read_size()` implementation to be correct.
2. The restrictions regarding `file_alignment` is **in** the implementation of `section_read_size()` so it should be checked there. This would make `section_read_size()` a fallible function, so its prototype and usage will need to account for this.

Reply:
so to clarify, are you suggesting some changes to the patch or it's ok?

Reply:
How about that:
```rust
fn section_read_size(section: &section_table::SectionTable, file_alignment: u32) -> Option<usize> {
....
    if file_alignment == 0 || file_alignment & (file_alignment - 1) != 0 {
        return None;
    }
....
}
```

Reply:
Another question is, if we check `file_aligment` in section_read_size returning Results/Option we have to change prototypes of these functions:

- is_in_range
- is_in_section
- find_offset

They will return result now, is it ok?

Reply:
That's what I meant by:
> This would make `section_read_size()` a fallible function, so its prototype and usage will need to account for this.

Reply:
To be honest, I'm starting to doubt that it is a good idea to check `file_alignment` so deep in code and return an error. There are lot's of places in code (20+) that use custom error messages via ok_or_else.  I don't know how to deal with it properly. Maybe we have to use anyhow crate, or somehow to combine errors from find_offset and custom error.

Maybe we have to check file_offset in place where it constructed or used first time?

Reply:
> To be honest, I'm starting to doubt that it is a good idea to check `file_alignment` so deep in code and return an error.

This is good for local reasoning, and for future maintenance of `section_read_size()`.

> There are lot's of places in code (20+) that use custom error messages via ok_or_else. I don't know how to deal with it properly. Maybe we have to use anyhow crate, or somehow to combine errors from find_offset and custom error.

That might make sense, but it seems excessive for what this PR is trying to fix. This might warrant another PR, though.

> Maybe we have to check file_alignment in place where it constructed or used first time?

This is bad for local reasoning, and for future maintenance of `section_read_size()`.